### DISCUSSION TITLE
[V3] Undocumented Breaking Change: layoutConfig from Array -> Object

### DISCUSSION THREAD

**Discussion by twellckhpx** [livewire/livewire#6571] [1. Help]
It seems the Documentation, specifically the "Upgrade Guide" does not document that layoutConfig is now an Object (use to be an array).

This is a breaking change for view macros (or other code) manipulating layoutConfig.

See https://github.com/livewire/livewire/commit/8300fdfd23882d6ac8f824a3d3735ce55e50a31b and https://github.com/livewire/livewire/commit/05038a87cf7ab3ed3fcfa3a2ccc4d94d98b282ff

**Comment by joshhanley** [comment]
@twellckhpx layoutConfig is internal, so it shouldn't impact users applications. If it impacted you, can your share your code so we can have a look? Thanks!

**Reply by twellckhpx** [reply]
Thanks for your comment @joshhanley!
As you said you usually might not interact with typical use cases of Livewire but can be common when you have custom View Macros.

It does look like the layoutConfig object can mostly be modified/merged through the defined macros registered in `src/Features/SupportPageComponents/SupportPageComponents.php` but we do access the object directly to get & merge some data (the latter could and probably should be done through the available macros).

In any case, I totally agree with you, it is a niche use case and issue. 
My main reason for bringing this up is that I did spend a bit of time trying to figure out what was causing those errors to be thrown after the upgrade from V2 to V3, until I finally found the commits mentioned above and I imagined it could affect other users.
